signer/src/crypto/secret_uri.rs

```diff
@@ -117,13 +118,20 @@ impl core::str::FromStr for SecretUri {
 }
 
 /// This is returned if `FromStr` cannot parse a string into a `SecretUri`.
-#[derive(Debug, Copy, Clone, PartialEq, Display)]
+#[derive(Debug, Copy, Clone, PartialEq)]
 pub enum SecretUriError {
     /// Parsing the secret URI from a string failed; wrong format.
-    #[display(fmt = "Invalid secret phrase format")]
     InvalidFormat,
 }
 
+impl Display for SecretUriError {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        match self {
+            SecretUriError::InvalidFormat => write!(f, "Invalid secret phrase format"),
+        }
+    }
+}
```

Comment:
I wonder if there is a small, reasonably maintained crate out there that can help us with this over manual impls, though happy doing it manually if not!

Reply:
i guess we can use Strum(uses syn 2.0) or we can just implement display by hand for now and hope that thiserror would start working in no-std or maybe try Using SNAFU for errors as it works in no-std(don't know if it was discussed before)